test/unit/mongo_client.test.js

```diff
@@ -876,4 +881,15 @@ describe('MongoOptions', function () {
       expect(client.options).to.have.property('mongoLoggerOptions').to.equal(expectedLoggingObject);
     });
   });
+
+  context('getAuthProvider', function () {
+    it('throws MongoInvalidArgumentError if provided authMechanism is not supported', function () {
+      const client = new MongoClient('mongodb://localhost:27017');
+      try {
+        client.s.authProviders.getOrCreateProvider('NOT_SUPPORTED');
```

Comment:
If this does not throw doesn't this test still pass?

Reply:
Thx for noticing!